internal/service/advancedclustertpf/resource_upgrade.go

```diff
@@ -6,7 +6,7 @@ import (
 )
 
 func getTenantUpgradeRequest(state, patch *admin.ClusterDescription20240805) *admin.LegacyAtlasTenantClusterUpgradeRequest {
-	if patch.ReplicationSpecs == nil {
+	if patch == nil || patch.ReplicationSpecs == nil {
```

Comment:
is this equivalent to len(patch.ReplicationSpecs()) == 0 ?

Reply:
it is a pointer, so cannot use length

Reply:
i think this works: 
```
if len(patch.GetReplicationSpecs()) == 0 {
```

GetReplicationSpecs() supports nil:
```
// GetReplicationSpecs returns the ReplicationSpecs field value if set, zero value otherwise
func (o *ClusterDescription20240805) GetReplicationSpecs() []ReplicationSpec20240805 {
	if o == nil || IsNil(o.ReplicationSpecs) {
		var ret []ReplicationSpec20240805
		return ret
	}
	return *o.ReplicationSpecs
}
```

Reply:
oh, it's even better to do:
```
if patch.HasReplicationSpecs() {
```

there is also `GetReplicationSpecsOk` to get the bool and the object.